readthedocs/rtd_tests/tests/test_api.py

```diff
@@ -354,6 +365,88 @@ def test_github_webhook(self, trigger_build):
         trigger_build.assert_has_calls(
             [mock.call(force=True, version=mock.ANY, project=self.project)])
 
+        client.post(
+            '/api/v2/webhook/github/{0}/'.format(self.project.slug),
+            {'ref': 'refs/heads/non-existent'},
+            format='json',
+        )
+        trigger_build.assert_has_calls(
+            [mock.call(force=True, version=mock.ANY, project=self.project)])
+
+        client.post(
+            '/api/v2/webhook/github/{0}/'.format(self.project.slug),
+            {'ref': 'refs/tags/v1.0'},
+            format='json',
+        )
+        trigger_build.assert_has_calls(
+            [mock.call(force=True, version=self.version_tag, project=self.project)])
+
+        client.post(
+            '/api/v2/webhook/github/{0}/'.format(self.project.slug),
+            {'ref': 'refs/heads/master'},
+            format='json',
+        )
+        trigger_build.assert_has_calls(
+            [mock.call(force=True, version=self.version, project=self.project)])
+
+    @mock.patch('readthedocs.restapi.views.integrations.WebhookMixin.get_response_push', return_value=None)
```

Comment:
Here the return value is None, since we only care to test if the ref value is parsed correctly, the other test covers the all flow

Reply:
Since what you want to test is the parsing, instead of writing the test in this way, why not avoid mocking this function and calling it _for real_ and assert the return value with the one that it's expected?

Like this:

    wm = WebhookMixin()
    assertEqual(wm._normalize_ref('refs/heads/master'), 'master')

It's clearer this way.

Reply:
I wasn't sure if that way was the correct one 😞.